ai_genomics/pipeline/description_embed/embed.py

```diff
@@ -0,0 +1,61 @@
+import click
+import json
+import logging
+import numpy as np
+import pandas as pd
+import os
+from sentence_transformers import SentenceTransformer
+from toolz import partition_all
+from torch import cuda
+
+
+logger = logging.getLogger(__name__)
+
+
+def embed(texts, model_name, chunk_size=10_000):
+    model = SentenceTransformer(model_name)
+    text_chunks = partition_all(chunk_size, texts)
+    embedding_chunks = [model.encode(tc) for tc in text_chunks]
+    return np.concatenate(embedding_chunks)
+
+
+@click.option(
+    "--directory",
+    prompt="Directory with data files.",
+)
+@click.option(
+    "--model-name",
+    prompt="Name of sentence transformer to use.",
+    default="allenai-specter",
+)
+@click.command()
+def run(directory, model_name):
+
```

Comment:
looks like you've saved .json files to s3, instead of saying someone needs to generate these files in the readme, you could:

```
cmd = f"aws s3 cp s3://ai-genomics/inputs/embedding/cb.json  {directory}/cb.json"
os.system(cmd)
logger.info(f"downloaded cb.json to {directory}")
cmd = f"aws s3 cp s3://ai-genomics/inputs/embedding/oa.json  {directory}/oa.json"
os.system(cmd)
logger.info(f"downloaded oa.json to {directory}")
cmd = f"aws s3 cp s3://ai-genomics/inputs/embedding/gtr.json  {directory}/gtr.json"
os.system(cmd)
logger.info(f"downloaded gtr.json to {directory}")
cmd = f"aws s3 cp s3://ai-genomics/inputs/embedding/pat.json  {directory}/pat.json"
os.system(cmd)
logger.info(f"downloaded pat.json to {directory}")
```

Reply:
To do this, you'd also have to set up your AWS credentials on the EC2 machine. It's a massive bodge. What I'd rather do is come back to this and write a flow for it that commissions batch machines automatically.